Sample data results in console errors in some locales

Due to data in /htm.java/src/main/java/org/numenta/nupic/datagen/rec-center-hourly.csv being in local time, the following error is appearing in the console:

`Exception in thread "Sensor Layer [r1:1] Thread" java.lang.IllegalStateException: org.joda.time.IllegalInstantException: Cannot parse "10/3/10 2:00": Illegal instant due to time zone offset transition (Australia/Sydney)
	at org.numenta.nupic.encoders.MultiEncoder.encodeIntoArray(MultiEncoder.java:105)
	at org.numenta.nupic.encoders.Encoder.encode(Encoder.java:625)
	at org.numenta.nupic.network.sensor.HTMSensor.input(HTMSensor.java:429)
	at org.numenta.nupic.network.sensor.HTMSensor.lambda$0(HTMSensor.java:362)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1812)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:294)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:169)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:300)
	at java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at org.numenta.nupic.network.sensor.HTMSensor$Copy.hasNext(HTMSensor.java:280)
	at java.util.Iterator.forEachRemaining(Iterator.java:115)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.numenta.nupic.network.Layer$5.run(Layer.java:2036)
Caused by: org.joda.time.IllegalInstantException: Cannot parse "10/3/10 2:00": Illegal instant due to time zone offset transition (Australia/Sydney)
	at org.joda.time.format.DateTimeParserBucket.computeMillis(DateTimeParserBucket.java:471)
	at org.joda.time.format.DateTimeParserBucket.computeMillis(DateTimeParserBucket.java:411)
	at org.joda.time.format.DateTimeFormatter.parseDateTime(DateTimeFormatter.java:882)
	at org.numenta.nupic.encoders.DateEncoder.parse(DateEncoder.java:446)
	at org.numenta.nupic.FieldMetaType.decodeType(FieldMetaType.java:98)
	at org.numenta.nupic.network.sensor.HTMSensor$InputMap.get(HTMSensor.java:306)
	at org.numenta.nupic.encoders.Encoder.getInputValue(Encoder.java:699)
	at org.numenta.nupic.encoders.MultiEncoder.encodeIntoArray(MultiEncoder.java:102)
	... 20 more
`

--- a/src/main/java/org/numenta/nupic/network/Layer.java
+++ b/src/main/java/org/numenta/nupic/network/Layer.java
@@ -21,6 +21,7 @@
  */
 package org.numenta.nupic.network;
 
+import java.lang.Thread.UncaughtExceptionHandler;
 import java.math.BigDecimal;
 import java.math.MathContext;
 import java.util.ArrayList;
@@ -2011,7 +2012,7 @@ public class Layer<T> implements Persistable {
      * Starts this {@code Layer}'s thread
      */
     protected void startLayerThread() {
-        (LAYER_THREAD = new Thread("Sensor Layer [" + getName() + "] Thread") {
+        LAYER_THREAD = new Thread("Sensor Layer [" + getName() + "] Thread") {
 
             @SuppressWarnings("unchecked")
             public void run() {
@@ -2044,7 +2045,16 @@ public class Layer<T> implements Persistable {
                     }
                 });
             }
-        }).start();
+        };
+        
+        LAYER_THREAD.setUncaughtExceptionHandler(new UncaughtExceptionHandler() {
+			
+			@Override
+			public void uncaughtException(Thread t, Throwable e) {
+				notifyError(new RuntimeException("Unhandled Exception in "+LAYER_THREAD.getName(),e));
+			}
+		});
+        LAYER_THREAD.start();
     }
     
     /**

--- a/src/test/java/org/numenta/nupic/network/PersistenceAPITest.java
+++ b/src/test/java/org/numenta/nupic/network/PersistenceAPITest.java
@@ -49,6 +49,7 @@ import java.util.stream.IntStream;
 import java.util.stream.Stream;
 
 import org.junit.AfterClass;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.numenta.nupic.FieldMetaType;
 import org.numenta.nupic.Parameters;
@@ -97,6 +98,13 @@ public class PersistenceAPITest extends ObservableTestBase {
     /** Printer to visualize DayOfWeek printouts - SET TO TRUE FOR PRINTOUT */
     private BiFunction<Inference, Integer, Integer> dayOfWeekPrintout = createDayOfWeekInferencePrintout(false);
     
+    
+    @BeforeClass
+    public static void beforeClass(){
+    	// Sample data contains datetimes that are invalid in some timezones due to DST.
+    	// If UTC is forced, then test runs should yield the same result regardless of timezone
+    	System.setProperty("user.timezone", "UTC");
+    }
 
     @AfterClass
     public static void cleanUp() {

